docs/proposals/kubelet-oci-runtime.md

```diff
@@ -0,0 +1,87 @@
+Kubelet OCI runtime integration
+===========================================================
+Authors: Mrunal Patel (@mrunalp), Vishnu Kannan (@vishh)
+
+## Abstract
+This proposal aims to introduce support for OCI Runtime Specification compatible runtimes in Kubelet.
+
+## Motivation
+Kubelet should support runc and OCI as a container runtime to support Open Standards. Runc is the OCI compliant
+implementation of the runtime specification for Linux. It has all the features in the OCI Runtime Spec, as well as
+runtime options to enable additional features. It should be possible to support pods using runc as it has all the
+primitives to support pods. In addition to runc, compatibility with OCI Specification will let kubernetes support other
+OCI compliant runtimes as well.
+
+## Design aspects
+The following subsections will discuss the various design aspects of the runc integration with kubelet. For the purposes
+of this doc, the kubelet runtime that will provide support for OCI compliant container runtimes will be referred to as
+`oci-runtime`.
+
+### Runtime API
+The implementation will make use of the kubelet runtime API (https://github.com/kubernetes/kubernetes/pull/25899). The
+kubelet will implement the client API and the container runtimes including the one for OCI will implement the server
+API. The separation of client/server allows for faster iteration on the runtime integration development.
+
+### Image management
+For the first release, oci-runtime will continue to use docker-engine for managing images.  The image management
+functionality will be separated from the runtime functionality so each could have different implementations which could
+potentially be switched.  It will be ideal to support the OCI image specification for images once it reaches v1.0. In
+addition to the docker image format, other packaging formats like tar files and docker image tar files, etc will also be
+supported. Image management requires quite a bit of flexibility in various fronts like transport, content verification,
+etc., and this proposal will not delve into those aspects. This proposal essentially opens up a door for easily
+extending support for alternate image formats.  Image managers are expected to expose the disk usage of each image and
+the overall usage of disk for all images.  To provide better security, it is possible to chown images of each pod to a
+separate uid and use user namespaces to have containers in that pod use that uid. This added security comes with a cost,
+because it will no longer be possible to share images between different pods and layering of images will no longer make
+any sense. 
```

Comment:
Yes and I would also like to suggest that these fetchers also be NOOPS and use shared storage.  Pulling images in production needs to support network stores.  If the image is available via NFS, Gluster, Cephs then you can bring up new nodes and instantly start containers.